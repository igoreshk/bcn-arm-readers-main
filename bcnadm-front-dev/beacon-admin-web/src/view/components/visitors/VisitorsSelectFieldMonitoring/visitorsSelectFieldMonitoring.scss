.secondGroup .visitorSelectFormControl {
  display: flex;
  align-items: stretch;
  height: 32px;
  width: 250px;
  margin-left: 38px;
  background-color: #6b6a6c;
  border-radius: 20px;
  cursor: pointer;

  @media screen and (max-width: 625px) {
    width: 300px;
    margin-bottom: 20px;
  }

  .visitorsSelect {
    color: #ffffff;
  }

  .select {
    margin: 0;
    padding: 5px 15px;
    display: flex;
    align-items: center;
    font-size: 14px;
  }

  .selectIcon {
    color: #ffffff;
    margin-right: 10px;
  }

}

