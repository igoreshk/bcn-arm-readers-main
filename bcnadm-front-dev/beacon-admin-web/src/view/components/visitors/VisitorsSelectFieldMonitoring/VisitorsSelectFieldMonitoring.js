import React from 'react';
import { FormControl, Select, MenuItem, ListItemIcon, Checkbox, ListItemText } from '@material-ui/core';
import PropTypes from 'prop-types';
import { useFilterVisitors } from '../useFilterVisitors';
import TitleSelect from '../TitleSelect';

import './visitorsSelectFieldMonitoring.scss';

const VisitorsSelectFieldMonitoring = ({ translate, visitors, changeListVisitorIds }) => {
  const { selectedVisitors, changeVisitors } = useFilterVisitors({ visitors, changeListVisitorIds });
  return (
    <FormControl className="visitorSelectFormControl">
      <Select
        multiple
        value={selectedVisitors}
        onChange={changeVisitors}
        renderValue={(value) => <TitleSelect listOfVisitors={value} translate={translate} />}
        disableUnderline
        displayEmpty
        classes={{
          root: 'visitorsSelect',
          select: 'select',
          icon: 'selectIcon'
        }}
      >
        <MenuItem value="" disabled>
          <span>{translate('monitoring.selectVisitors')}</span>
        </MenuItem>
        {visitors.length !== 0 ? (
          <MenuItem data-button-all>
            <ListItemIcon>
              <Checkbox
                checked={visitors.length > 0 && selectedVisitors.length === visitors.length}
                indeterminate={selectedVisitors.length > 0 && selectedVisitors.length < visitors.length}
              />
            </ListItemIcon>
            <ListItemText primary={translate('monitoring.allVisitors')} />
          </MenuItem>
        ) : (
          ''
        )}
        {visitors.map((visitor) => (
          <MenuItem name="visitor" key={visitor.entityId} value={visitor} className="visitorItem">
            {visitor.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

VisitorsSelectFieldMonitoring.propTypes = {
  visitors: PropTypes.arrayOf(
    PropTypes.shape({
      entityId: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  changeListVisitorIds: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default VisitorsSelectFieldMonitoring;
