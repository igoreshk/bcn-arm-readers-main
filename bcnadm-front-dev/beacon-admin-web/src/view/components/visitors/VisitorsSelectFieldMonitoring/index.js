import VisitorsSelectFieldMonitoring from './VisitorsSelectFieldMonitoring';
export default VisitorsSelectFieldMonitoring;
