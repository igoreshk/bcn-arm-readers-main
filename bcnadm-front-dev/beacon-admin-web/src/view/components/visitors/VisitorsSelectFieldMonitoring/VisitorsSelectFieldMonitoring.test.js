import React from 'react';
import { shallow } from 'enzyme';
import { Select } from '@material-ui/core';
import VisitorsSelectFieldMonitoring from './VisitorsSelectFieldMonitoring';

const props = {
  visitors: [
    {
      entityId: '5fc8bf048dfe557a0c9a9191',
      name: 'George'
    }
  ],
  changeListVisitorIds: jest.fn(),
  translate: jest.fn()
};
const value = [{ entityId: '61c6fc23d0786b15b4caff5b', name: 'George' }];
const wrapper = shallow(<VisitorsSelectFieldMonitoring {...props} />);

describe('VisitorsSelectFieldMonitoring', () => {
  it('Selecting a new visitor should call translate', () => {
    wrapper.find(Select).simulate('click', { target: { value } });
    expect(props.translate).toHaveBeenCalled();
  });

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
