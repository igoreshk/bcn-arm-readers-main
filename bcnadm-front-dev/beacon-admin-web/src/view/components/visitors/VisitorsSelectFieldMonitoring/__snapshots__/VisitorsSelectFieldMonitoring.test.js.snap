// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisitorsSelectFieldMonitoring Check matching snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="visitorSelectFormControl"
>
  <WithStyles(ForwardRef(Select))
    classes={
      Object {
        "icon": "selectIcon",
        "root": "visitorsSelect",
        "select": "select",
      }
    }
    disableUnderline={true}
    displayEmpty={true}
    multiple={true}
    onChange={[Function]}
    renderValue={[Function]}
    value={Array []}
  >
    <WithStyles(ForwardRef(MenuItem))
      disabled={true}
      value=""
    >
      <span />
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      data-button-all={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(Checkbox))
          checked={false}
          indeterminate={false}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText)) />
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="visitorItem"
      key="5fc8bf048dfe557a0c9a9191"
      name="visitor"
      value={
        Object {
          "entityId": "5fc8bf048dfe557a0c9a9191",
          "name": "George",
        }
      }
    >
      George
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</WithStyles(ForwardRef(FormControl))>
`;
